src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -243,7 +257,19 @@ public void FileModifier_06()
     {
         UsingNode($$"""
             partial file record struct C { }
-            """);
+            """,
+            // (1,1): error CS1031: Type expected
+            // partial file record struct C { }
+            Diagnostic(ErrorCode.ERR_TypeExpected, "partial").WithLocation(1, 1),
+            // (1,1): error CS1525: Invalid expression term 'partial'
+            // partial file record struct C { }
+            Diagnostic(ErrorCode.ERR_InvalidExprTerm, "partial").WithArguments("partial").WithLocation(1, 1),
+            // (1,1): error CS1003: Syntax error, ',' expected
+            // partial file record struct C { }
+            Diagnostic(ErrorCode.ERR_SyntaxError, "partial").WithArguments(",").WithLocation(1, 1),
+            // (1,21): error CS1002: ; expected
+            // partial file record struct C { }
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "struct").WithLocation(1, 21));
```

Comment:
Definitely not the most pleasant diagnostics. The improvement here is tracked by https://github.com/dotnet/roslyn/issues/61025.

Reply:
Fixed.